synapseclient/api/agent_services.py

```diff
@@ -0,0 +1,259 @@
+"""This module is responsible for exposing the services defined at:
+<https://rest-docs.synapse.org/rest/#org.sagebionetworks.repo.web.controller.AgentController>
+"""
+
+import asyncio
+import json
+from typing import TYPE_CHECKING, Any, Dict, Optional
+
+if TYPE_CHECKING:
+    from synapseclient import Synapse
+
+from synapseclient.core.exceptions import SynapseTimeoutError
+
+
+async def register_agent(
+    cloud_agent_id: str,
+    cloud_alias_id: Optional[str] = None,
+    synapse_client: Optional["Synapse"] = None,
+) -> Dict[str, Any]:
+    """
+    Registers an agent with Synapse OR gets existing agent registration.
+
+    Arguments:
+        cloud_agent_id: The cloud provider ID of the agent to register.
+        cloud_alias_id: The cloud provider alias ID of the agent to register.
+                        In the Synapse API, this defaults to 'TSTALIASID'.
+        synapse_client: If not passed in and caching was not disabled by
+                `Synapse.allow_client_caching(False)` this will use the last created
+                instance from the Synapse class constructor.
+
+    Returns:
+        The registered agent matching
+            <https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/agent/AgentRegistration.html>
+    """
+    from synapseclient import Synapse
+
+    client = Synapse.get_client(synapse_client=synapse_client)
+
+    # Request matching <https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/agent/AgentRegistrationRequest.html>
+    request = {
+        "awsAgentId": cloud_agent_id,
+        "awsAliasId": cloud_alias_id if cloud_alias_id else None,
```

Comment:
Will this request work if you pass in None? My assumption is that the key in the http body wouldn't be sent if there is no value

Reply:
It will work because the `awsAliasId` is optional in the [API endpoint](https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/agent/AgentRegistrationRequest.html) itself, with a set default value. I guess there isn't really a reason to not just pass that default value here though.

Reply:
It's optional in the API, but i wasn't certain if it would send the literal string "None" here, or if it drops the field from the body, or it leaves it as blank when sending the request

Reply:
Oh I see. It leaves it blank as far as I can tell

Reply:
Double check the behavior here. Blank does have a different meaning than the absence of the key (null).

It might not affect this api but i know I other areas of the code none keys are getting popped out of the body before the request gets sent

Reply:
Would it make more sense to just pass the API default as the default here too and not have to worry about it?

Reply:
If we send the default in the python client, and the default changes server side - Anyone staying on that version of the client will continue to send in (old) defaults. I would rather we don't send the key at all and only send it if it's a non-None value.

That way if the default changes (server side) the client side code has no changes.